Mathlib/NumberTheory/Cyclotomic/PrimitiveRoots.lean

```diff
@@ -189,6 +189,85 @@ theorem finrank (hirr : Irreducible (cyclotomic n K)) : finrank K L = (n : ℕ).
     (zeta_spec n K L).minpoly_eq_cyclotomic_of_irreducible hirr, natDegree_cyclotomic]
 #align is_cyclotomic_extension.finrank IsCyclotomicExtension.finrank
 
+variable {L} in
+/-- If `L` contains both a primitive `p`-th root of unity and `q`-th root of unity, and
+`Irreducible (cyclotomic (lcm p q) K)` (in particular for `K = ℚ`), then the `finrank K L` is at
+least `(lcm p q).totient`. -/
+theorem _root_.IsPrimitiveRoot.lcm_totient_le_finrank [FiniteDimensional K L] {p q : ℕ} {x y : L}
+    (hx : IsPrimitiveRoot x p) (hy : IsPrimitiveRoot y q)
+    (hirr : Irreducible (cyclotomic (Nat.lcm p q) K)) :
+    (Nat.lcm p q).totient ≤ FiniteDimensional.finrank K L := by
+  rcases Nat.eq_zero_or_pos p with (rfl | hppos)
+  · simp
+  rcases Nat.eq_zero_or_pos q with (rfl | hqpos)
+  · simp
+  let z := x ^ (p / factorizationLCMLeft p q) * y ^ (q / factorizationLCMRight p q)
+  let k := PNat.lcm ⟨p, hppos⟩ ⟨q, hqpos⟩
+  have : IsPrimitiveRoot z k := hx.pow_mul_pow_lcm hy hppos.ne' hqpos.ne'
+  haveI := IsPrimitiveRoot.adjoin_isCyclotomicExtension K this
+  convert Submodule.finrank_le (Subalgebra.toSubmodule (adjoin K {z}))
+  replace hirr : Irreducible (cyclotomic k K) := hirr
```

Comment:
This is changing the `Nat.lcm` to `PNat.lcm` and using `k`? I'd personally prefer something more explicit than `replace`

Reply:
Done